app/presenters/task_list/collection.rb

```diff
@@ -32,41 +30,28 @@ class Collection < SimpleDelegator
       ]
     }.freeze
 
-    def initialize(view, crime_application:)
-      @view = view
+    def initialize(crime_application:)
       @crime_application = crime_application
-
-      super(collection)
-    end
-
-    def render
-      tag.ol class: 'moj-task-list' do
-        safe_join(map(&:render))
-      end
     end
 
     def completed
-      all_tasks.select(&:completed?)
+      all_tasks.select { |task| task.status.completed? }
     end
 
     def applicable
-      all_tasks.reject(&:not_applicable?)
-    end
-
-    private
-
-    def all_tasks
-      @all_tasks ||= map(&:items).flatten
+      all_tasks.reject { |task| task.status.not_applicable? }
     end
 
```

Comment:
I've refactored the section renderer into two distinct parts: a TaskLit::Section (retaining the same class name as the old renderer) and a new TaskList::SectionComponent.

This change was driven by the lack of dedicated models for the TaskList classes. These classes currently encapsulate significant business logic, and could be used elsewhere. I believe moving towards dedicated models for Section, Task etc will improve things.

I've not moved them from the presenter folder, but I think we could. For now I've just separated business logic from view-related concerns.